src/erpbrasil/base/fiscal/recopi.py

```
@@ -0,0 +1,162 @@
+    6,
+    5,
+    4,
+    3,
+    2,
+    1,
+]
+
+
+def is_valid_recopi(recopi):
+    if _check_recopi_size(recopi):
+        if _check_recopi_format(recopi):
+            return True
+    return False
+
+
+def generate_valid_recopi():
+    recopi_date = _generate_recopi_date()
+    recopi_hour = _generate_recopi_hour()
+    recopi_four_digits = _generate_recopi_four_digits()
+    recopi_verification_digits = _generate_recopi_verification_digits(
+        recopi_date + recopi_hour + recopi_four_digits
+    )
+    recopi = recopi_date + recopi_hour + recopi_four_digits + recopi_verification_digits
+    return recopi
+
+
+def _check_recopi_size(recopi):
+    if len(recopi) == 20:
+        return True
+    return False
+
+
+def _check_recopi_format(recopi):
+    return (
+        _check_recopi_date(recopi)
+        and _check_recopi_time(recopi)
+        and _check_recopi_verification_digits(recopi)
+    )
+
+
+def _check_recopi_date(recopi):
+    recopi_year, recopi_month, recopi_day = get_recopi_date(recopi)
+    try:
+        dt.date(
+            year=recopi_year, month=recopi_month, day=recopi_day
+        )  # _check if it's a calendar date
+        if recopi_year > CURRENT_YEAR:
+            return True
+        elif recopi_year == CURRENT_YEAR and recopi_month > CURRENT_MONTH:
+            return True
+        elif recopi_year < 2013:
+            return True
+
+        return False
+    except:
+        return False
+
+
+def get_recopi_date(recopi):
+    recopi_year = int(recopi[0:4])
+    recopi_month = int(recopi[4:6])
+    recopi_day = int(recopi[6:8])
+    return recopi_year, recopi_month, recopi_day
+
+
+def _check_recopi_time(recopi):
+    recopi_hour, recopi_minute, recopi_second = get_recopi_time(recopi)
+    try:
+        dt.datetime(
+            year=2003,
+            month=4,
+            day=21,
+            hour=recopi_hour,
+            minute=recopi_minute,
+            second=recopi_second,
+        )
+        return True
+    except:
```

Corrigir o uso do except
Não utilize o bare except, pois ele captura qualquer exceção.
Exemplo: except Exception

https://peps.python.org/pep-0008/#programming-recommendations